File: front/types/tokenizer.ts
// eslint-disable-next-line @typescript-eslint/no-unused-vars
const TIKTOKEN_TOKENIZER_BASE = [
  "o200k_base",
  "cl100k_base",
  "p50k_base",
  "r50k_base",
  "anthropic_base",
] as const;

type TiktokenTokenizerBase = (typeof TIKTOKEN_TOKENIZER_BASE)[number];

// eslint-disable-next-line @typescript-eslint/no-unused-vars
const SENTENCEPIECE_BASE = ["model_v1", "model_v2", "model_v3"] as const;

type SentencePieceTokenizerBase = (typeof SENTENCEPIECE_BASE)[number];

export type TokenizerConfig =
  | { type: "tiktoken"; base: TiktokenTokenizerBase }
  | { type: "sentence_piece"; base: SentencePieceTokenizerBase };

File: core/src/types/tokenizer.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum TiktokenTokenizerBase {
    O200kBase,
    Cl100kBase,
    P50kBase,
    R50kBase,
    AnthropicBase,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum SentencePieceTokenizerBase {
    ModelV1,
    ModelV2,
    ModelV3,
}

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
#[serde(tag = "type", rename_all = "snake_case")]
pub enum TokenizerConfig {
    Tiktoken { base: TiktokenTokenizerBase },
    SentencePiece { base: SentencePieceTokenizerBase },
}
